File: Daily Challanges/2026-05-May/2026-05-06.py
def get_allergen_friendly_meals(meals, allergens):
    safe_meals = []
    allergens = set(allergens)
    for meal in meals:
        for allergen in meal[1]:
            if allergen in allergens:
                break;
        else:
            safe_meals.append(meal[0])
    return safe_meals

print(get_allergen_friendly_meals([["pasta", ["wheat", "milk"]], ["salad", ["nuts"]]], ["milk"]))
print(get_allergen_friendly_meals([["steak", ["soy"]], ["fried rice", []], ["fish tacos", ["fish", "wheat"]], ["chicken parmesan", ["wheat", "milk"]]], ["soy", "fish"]))

File: Daily Challanges/2026-05-May/2026-05-06.js
function getAllergenFriendlyMeals(meals, allergens) {
  let safeMeals = [];
  const avoid = new Set(allergens);
  for (const [mealName, mealAllergens] of meals) {
    const hasBadAllergen = mealAllergens.some((allergen) =>
      avoid.has(allergen),
    );
    if (!hasBadAllergen) safeMeals.push(mealName);
  }
  return safeMeals;
}

console.log(
  getAllergenFriendlyMeals(
    [
      ["pasta", ["wheat", "milk"]],
      ["salad", ["nuts"]],
    ],
    ["milk"],
  ),
);
console.log(
  getAllergenFriendlyMeals(
    [
      ["steak", ["soy"]],
      ["fried rice", []],
      ["fish tacos", ["fish", "wheat"]],
      ["chicken parmesan", ["wheat", "milk"]],
    ],
    ["soy", "fish"],
  ),
);
